src/Features/CSharp/Portable/SignatureHelp/InvocationExpressionSignatureHelpProviderBase_MethodGroup.cs

```diff
@@ -53,9 +53,12 @@ internal partial class InvocationExpressionSignatureHelpProviderBase
                     throughType = semanticModel.GetTypeInfo(throughExpression, cancellationToken).Type;
                 }
 
-                var includeInstance = !throughExpression.IsKind(SyntaxKind.IdentifierName) ||
+                // SyntaxKind.IdentifierName is for basic case, e.g. "MyClass.MyStaticMethod(...)"
+                // SyntaxKind.SimpleMemberAccessExpression is for not imported types, e.g. "MyNamespace.MyClass.MyStaticMethod(...)"
+                // SyntaxKind.PredefinedType is for built-in types, e.g. "string.Equals(...)"
+                var includeInstance = !throughExpression.IsKind(SyntaxKind.IdentifierName, SyntaxKind.SimpleMemberAccessExpression, SyntaxKind.PredefinedType) ||
```

Comment:
i don't get the identifiername/memberaccess cases.  if i do `someInstance.` i should get instance members right?

Reply:
`IdentifierName` was there even bedore this PR) This part of the condition checks only the left-hand side of the expression, e.g. `[|someInstance|].NonStaticMember()`. As I described in the comment, `IdentifierName` is for the case of `[|ClassName|].StaticMember()`, `ClassName` is `IdentifierName` here. Same for `[|NamespaceName.ClassName|].StaticMember()` where `NamespaceName.ClassName` is a `SimpleMemberAccessExpression` and `[|string|].Equals()` where `string` is `PredefinedType`. However, this condition doesn't filter cases like `[|someInstance.SomeProperty|].NonStaticMember()` where `someInstance.SomeProperty` is a `SimpleMemberAccessExpression`, because other parts of the whole expression are responsible for this. When I initially added this logic, none of existing tests failed.

Reply:
got it.  i misread the condition.  lots of nots and ors break my brain.  this LGTM.  thanks!